keanu-project/src/main/java/io/improbable/keanu/tensor/TensorShape.java

```diff
@@ -182,5 +182,24 @@ public static boolean isScalar(int[] shape) {
         return newShape;
     }
 
+    /**
+     * Writes a lower rank shape over a higher rank shape, starting from the right.
+     *
+     * e.g: high rank shape = [1, 2, 2, 1]
+     *      low rank shape = [1, 4]
+     *
+     * Result after copy = [1, 2, 1, 4]
+     *
+     * @param higherRankShape source shape that will get written over
+     * @param lowerRankShape  shape to write
+     * @return the high rank  shape with the lower rank shape inserted on top of it
+     */
+    public static int[] copyLowRankOverHighRank(int[] higherRankShape, int[] lowerRankShape) {
+        int[] partialShape = Arrays.copyOf(higherRankShape, higherRankShape.length);
```

Comment:
The terms `partial` and `of` relate to partial derivatives. These should be renamed to fix in the context of TensorShape since it lives it that class.

Reply:
Good spot - will update